moshi/src/main/java/com/squareup/moshi/JsonAdapter.java

```diff
@@ -37,7 +37,9 @@
   }
 
   public final @Nullable T fromJson(String string) throws IOException {
-    return fromJson(new Buffer().writeUtf8(string));
+    try (Buffer buffer = new Buffer()) {
```

Comment:
This is also weird and seems unnecessary. Calling `close()` on a `Buffer` [is a no-op][1] so there's nothing gained by doing this.

 [1]: https://github.com/square/okio/blob/20e259c08af6c96508387540a5ea53f89b1d01f5/okio/src/main/java/okio/Buffer.java#L1530-L1531